programs/clearing_house/src/math/liquidation.rs

```diff
@@ -295,3 +299,126 @@ pub fn calculate_cumulative_deposit_interest_delta_to_resolve_bankruptcy(
         .safe_div(total_deposits)
         .or(Ok(0))
 }
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct DeleverageUserStats {
+    pub base_asset_amount: i64,
+    pub quote_asset_amount: i64,
+    pub quote_entry_amount: i64,
+    pub free_collateral: i128,
+}
+
+pub fn calculate_perp_market_deleverage_payment(
+    loss_to_socialize: i128,
+    deleverage_user_stats: DeleverageUserStats,
+    market: &PerpMarket,
+    oracle_price: i64,
+) -> ClearingHouseResult<i128> {
+    let base_amount =
+        (-market.amm.base_asset_amount_short).safe_add(market.amm.base_asset_amount_long)?;
+
+    let cost_basis_above_mean = if deleverage_user_stats.base_asset_amount != 0 {
```

Comment:
nit: is `above` right terminology now? because `above mea`n is good or bad depending on long versus shorts